assembly/src/parsers/ast/io_ops.rs

```diff
@@ -34,14 +34,13 @@ pub(super) fn parse_adv_push(op: &Token) -> Result<Node, AssemblyError> {
     validate_operation!(op, "adv_push", 1);
 
     let param = parse_param::<u32>(op, 1)?;
-    Ok(Node::Instruction(Instruction::Locaddr(Felt::from(param))))
+    Ok(Node::Instruction(Instruction::AdvPush(param as u16)))
```

Comment:
Rather than parsing this as `u32` and then casting, we should parse this as `u8` to begin with.

Also, we should probably validate that the value is not greater than 16 somewhere.

Reply:
I've copied the validation helper function from the old parser to the new io_ops helper functions section. I'm not sure how the new parser is planned to be structured so please let me know if this should be put somewhere else.

Reply:
Thank you! I would probably put it into `/ast/mod.rs` next to `parse_param()` function. Also, we should probably make it generic so that we can parse not just `u8`'s with it in the future. The signature could probably look something like this:
```Rust
fn parse_checked_param<I: core::str::FromStr + Ord + Display>(
    op: &Token,
    param_idx: usize,
    lower_bound: I,
    upper_bound: I,
) --> Result<I, AssemblyError>
```

Reply:
Thanks. Done.